gcc version 8 compiler warning in msc_device.c

When building circuitpython **with arm-none-eabi-gcc 8.2.1** (arm toolchain 8-2018-q4), I get this fatal warning:
```
../../lib/tinyusb/src/class/msc/msc_device.c: In function 'proc_builtin_scsi':
../../lib/tinyusb/src/class/msc/msc_device.c:273:7: error: 'strncpy' output truncated before terminating nul copying 8 bytes from a string of the same length [-Werror=stringop-truncation]
       strncpy((char*) inquiry_rsp.vendor_id  , CFG_TUD_MSC_VENDOR     , sizeof(inquiry_rsp.vendor_id));
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../lib/tinyusb/src/class/msc/msc_device.c:274:7: error: 'strncpy' output truncated before terminating nul copying 16 bytes from a string of the same length [-Werror=stringop-truncation]
       strncpy((char*) inquiry_rsp.product_id , CFG_TUD_MSC_PRODUCT    , sizeof(inquiry_rsp.product_id));
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```
I tried a few ways to fix it, first using `memcpy()` and I think I remember that `CFG_TUD_MSC_PRODUCT` and `VENDOR` are not the same length as `inquiry_rsp.product_id` and `.vendor_id` correspondingly, which causes various issues.

Only gcc version 8 is catching this issue. We haven't moved up to gcc 8 yet, but we will sooner or later.

--- a/src/class/msc/msc_device.c
+++ b/src/class/msc/msc_device.c
@@ -265,14 +265,20 @@ int32_t proc_builtin_scsi(msc_cbw_t const * p_cbw, uint8_t* buffer, uint32_t buf
           .is_removable         = 1,
           .version              = 2,
           .response_data_format = 2,
-          .vendor_id            = "Adafruit",
-          .product_id           = "Feather52840",
-          .product_rev          = "1.0"
+          .vendor_id            = "        ",
+          .product_id           = "                ",
+          .product_rev          = "    ",
       };
+      size_t len;
 
-      strncpy((char*) inquiry_rsp.vendor_id  , CFG_TUD_MSC_VENDOR     , sizeof(inquiry_rsp.vendor_id));
-      strncpy((char*) inquiry_rsp.product_id , CFG_TUD_MSC_PRODUCT    , sizeof(inquiry_rsp.product_id));
-      strncpy((char*) inquiry_rsp.product_rev, CFG_TUD_MSC_PRODUCT_REV, sizeof(inquiry_rsp.product_rev));
+      #define _min(a,b) ((a) < (b) ? (a) : (b))
+      len = strlen(CFG_TUD_MSC_VENDOR);
+      memcpy(inquiry_rsp.vendor_id  , CFG_TUD_MSC_VENDOR     , _min(len, sizeof(inquiry_rsp.vendor_id)));
+      len = strlen(CFG_TUD_MSC_PRODUCT);
+      memcpy(inquiry_rsp.product_id , CFG_TUD_MSC_PRODUCT    , _min(len, sizeof(inquiry_rsp.product_id)));
+      len = strlen(CFG_TUD_MSC_PRODUCT_REV);
+      memcpy(inquiry_rsp.product_rev, CFG_TUD_MSC_PRODUCT_REV, _min(len, sizeof(inquiry_rsp.product_rev)));
+      #undef _min
 
       ret = sizeof(inquiry_rsp);
       memcpy(buffer, &inquiry_rsp, ret);

